### DISCUSSION TITLE
Is it possible to use @defer and @stream directives?

### DISCUSSION THREAD

**Discussion by myshootko** [hayes/pothos#1130] [Q&A]
Hello everyone,

I would like to ask if there is some way how to use @defer and @stream GraphQL directives with Pothos version 3.41.0 and Directives plugin. 

I tried this at first, but it doesn't seem to work:

```// Define the @defer directive
export const deferDirective = new GraphQLDirective({
  name: 'defer',
  locations: ['FIELD', 'FRAGMENT_SPREAD', 'INLINE_FRAGMENT'] as DirectiveLocation[],
  args: {
    // if: { type: GraphQLBoolean },
    // label: { type: GraphQLString },
  },
});

// Define the @stream directive
export const streamDirective = new GraphQLDirective({
  name: 'stream',
  locations: ['FIELD'] as DirectiveLocation[],
  args: {
    //   if: { type: GraphQLBoolean },
    //   initialCount: { type: GraphQLInt },
  },
});

builder.queryType({
  directives: [deferDirective, streamDirective],
});```

Any ideas would be appreciated.

**Comment by hayes** [comment]
You don't need anything in Pothos to use these, as long as your server (eg yoga) supports them, it should just work out of the box

**Reply by myshootko** [reply]
I am using Apollo server + Apollo client and even in the studio I am getting ```GraphQLError: Unknown directive "@defer"``` when trying to run query with @defer.

Previously we were using Nexus as a code first library and there we had to define these two first to be even included in schema like this:

```
directive @defer(if: Boolean, label: String) on FRAGMENT_SPREAD | INLINE_FRAGMENT

directive @stream(if: Boolean, initialCount: Int = 0, label: String) on FIELD
```

**Reply by hayes** [reply]
Defer and stream are built in directives and don't need to be part of the schema.   Id double check that you are using the latest version of graphql and any Apollo server packages.  I would also try using something like pnpm/yarn resolutions to make sure you only have 1 copy of graphql and that it's the latest version

**Reply by myshootko** [reply]
ok, thank you. will try to resolve it somehow.

**Reply by toinhao1** [reply]
@hayes with Apollo Server it requires the directives be defined in the schema: https://www.apollographql.com/docs/apollo-server/workflow/requests/#incremental-delivery-experimental

**Reply by hayes** [reply]
I see.  You can pass directives for the schema to use in 

```ts
builder.toSchema({
  directives: [
    ...specifiedDirectives,
    GraphQLDeferDirective,
    GraphQLStreamDirective
  }
})
```

**Reply by toinhao1** [reply]
Awesome, that works perfectly. Thank you!

**Reply by myshootko** [reply]
Thanks for your help. Now it works :)
